dali/base/dautils.cpp

```diff
@@ -456,28 +456,62 @@ void CDfsLogicalFileName::normalizeName(const char *name, StringAttr &res, bool
     res.set(str);
 }
 
-bool isExternalFile(const char *name)
+bool isExternalFile(const char *name, StringAttr &lfn, bool strict, unsigned &tailpos)
 {
     // TODO Should check the name is a valid OS filename
     if ('~' == *name) // allowed 1 leading ~
+    {
         name++;
+        if (!strict)
+            skipSp(name);
+    }
 
     bool retVal = memicmp(name,EXTERNAL_SCOPE "::",sizeof(EXTERNAL_SCOPE "::")-1)==0;
-    const char *s=strstr(name,"::");
-    if (retVal && s)
+    if (retVal)
     {
-        const char *s1 = s+2;
-        const char *ns1 = strstr(s1,"::");
-        if (!ns1)
-            retVal = false;
-        else
+        lfn.clear();
+        StringBuffer str;
+        const char *s=strstr(name,"::");
```

Comment:
It think this is redundant.. "::" must exist, it's been checked in the memicmp above.
You still need to skip over EXTERNAL_SCOPE here though.